proxy/helpers.go

```diff
@@ -0,0 +1,33 @@
+package proxy
+
+import (
+	"encoding/json"
+)
+
+// An ethereum request abstraction. 
+// Only need it for one method, though.
+type ethRequest struct {
+	// ChainId to indicate which Ethereum-based network
+	ID int `json:"id"` 
+	// Indicates which version of JSON RPC to use
+	// Since all networks support JSON RPC 2.0,
+	// this attribute is a constant
+	JSONRPC string `json:"jsonrpc"`
+	// Which method you want to call
+	METHOD string `json:"method"`
+	// Params for the method you want to call
+	Params []string `json:"params"`
+}
+
+// Takes a chainId and signed transaction data as parameters
+// Returns a JSON encoding of the request
+func marshalRequest(id int, params []string) ([]byte, error) {
```

Comment:
It is more idiomatic to have a function that initiates a new data structure and then to use the `json.Marshal()` at will. This instead of having a `mashalRequest()` function that takes care of it all.

More concretely, instead of defining `mashalRequest(id, params)`, define a `NewRequest(id, params)` which returns an initialised data structure and then use `json.Marshal(...)` when needed.

Not a big deal though, it will work as is.

Reply:
Due to this suggestion, I have simply moved `NewRequest` to `proxy.go` and removed `helpers.go`.